src/collection-browser.ts

```diff
@@ -1211,10 +1211,19 @@ export class CollectionBrowser
       this.preloadCollectionNames(results);
       this.updateDataSource(pageNumber, results);
     }
-    if (results.length < this.pageSize) {
+
+    // When we reach the end of the data, we can't always immediately cap off the
+    // tile count, because there may be previous pages that haven't finished loading
+    // yet. So we flag the end of the data, but let the final page load set the count.
+    // This avoids race conditions dependent on the order pages arrive in.
+    if (results.length < this.pageSize || this.endOfDataReached) {
       this.endOfDataReached = true;
-      // this updates the infinite scroller to show the actual size
-      if (this.infiniteScroller) {
+      // This updates the infinite scroller to show the actual size
+      // (provided there are no other pages still being fetched)
+      if (
+        this.infiniteScroller &&
```

Comment:
Q: do we the need to check if `this.infiniteScroller`? feels like the `infiniteScroller` should be absolute..

Reply:
Good question! The infinite scroller should always exist _except_ when a "no query"/"no results" placeholder is shown. And that placeholder should be cleared before we start fetching a page. So I don't _think_ it's possible to reach this point without the infinite scroller existing.

It's possible this guard was just added out of an abundance of caution, since `this.infiniteScroller` is defined by a `@query` decorator.